src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala

```diff
@@ -410,6 +387,89 @@ object P2PKScriptSignature extends Factory[P2PKScriptSignature] {
   }
 }
 
+sealed trait CLTVScriptSignature extends ScriptSignature {
+  def scriptSig : ScriptSignature = ScriptSignature(hex)
+
+  override def signatures : Seq[ECDigitalSignature] = scriptSig.signatures
+
+  override def hex = scriptSig.hex
+}
+
+object CLTVScriptSignature extends Factory[CLTVScriptSignature] {
+  private case class CLTVScriptSignatureImpl(override val hex : String) extends CLTVScriptSignature
+
+  override def fromBytes(bytes : Seq[Byte]) : CLTVScriptSignature = {
+    val hex = BitcoinSUtil.encodeHex(bytes)
+    fromHex(hex)
+  }
+
+  override def fromHex(hex : String) : CLTVScriptSignature = {
+    CLTVScriptSignatureImpl(hex)
+  }
+
+  def apply(scriptSig : ScriptSignature) : CLTVScriptSignature = {
+    fromHex(scriptSig.hex)
+  }
+
+  /**
+    * Creates a CLTVScriptSignature out the [[ScriptPubKey]] we are satisfying, a sequence of [[ECDigitalSignature]], and a sequence
+    * of [[ECPublicKey]] needed to satisfy the scriptPubKey. If a [[P2SHScriptPubKey]] is provided, a redeemScript must also be provided.
+    * @return
+    */
+  def apply(scriptPubKey: ScriptPubKey, sigs : Seq[ECDigitalSignature], pubKeys : Seq[ECPublicKey]) : CLTVScriptSignature = scriptPubKey match {
+    case p2pkScriptPubKey : P2PKScriptPubKey => CLTVScriptSignature(P2PKScriptSignature(sigs.head))
+    case p2pkhScriptPubKey : P2PKHScriptPubKey => CLTVScriptSignature(P2PKHScriptSignature(sigs.head, pubKeys.head))
+    case multiSigScriptPubKey : MultiSignatureScriptPubKey => CLTVScriptSignature(MultiSignatureScriptSignature(sigs))
+    case cltvScriptPubKey : CLTVScriptPubKey => apply(cltvScriptPubKey.scriptPubKeyAfterCLTV, sigs, pubKeys)
+    case csvScriptPubKey : CSVScriptPubKey => apply(csvScriptPubKey.scriptPubKeyAfterCSV, sigs, pubKeys)
+    case EmptyScriptPubKey => CLTVScriptSignature(EmptyScriptSignature)
+    case _ : NonStandardScriptPubKey | _ : P2SHScriptPubKey => throw new IllegalArgumentException("A NonStandardScriptSignature or P2SHScriptSignature cannot be" +
+      "the underlying scriptSig in a CLTVScriptSignature. Got: " + this)
```

Comment:
This is an incorrect error message, `this` would return the object we currently in, in this case `CLTVScriptSignature`. You want to return the scriptPubKey we are pattern matching on. You can do something like this 

```scala

case x @ (_ : NonStandardScriptPubKey | _ : P2SHScriptPubKey) => throw new IllegalArgumentException("A NonStandardScriptSignature or P2SHScriptSignature cannot be" +
 +      "the underlying scriptSig in a CLTVScriptSignature. Got: " + x)
```